lms/djangoapps/grades/tests/test_transformer.py

```diff
@@ -102,6 +102,98 @@ def build_course_with_problems(self, data='<problem></problem>', metadata=None):
             }
         ])
 
+    def build_complicated_hypothetical_course(self):
+        """
+        Create a test course with a very odd structure as a stress-test for various methods.
+
+        Current design is to test containing_subsection logic in collect_unioned_set_field.
+        I can't reasonably draw this in ascii art (due to intentional complexities), so here's an overview:
+            We have 1 course, containing 1 chapter, containing 2 subsections.
+
+            From here, it starts to get hairy. Call our subsections A and B.
+            Subsection A contains 3 verticals (call them 1, 2, and 3), and another subsection (C)
+            Subsection B contains vertical 3 and subsection C
+            Subsection C contains 1 problem (b)
+            Vertical 1 contains 1 vertical (11)
+            Vertical 2 contains no children
+            Vertical 3 contains no children
+            Vertical 11 contains 1 problem (aa) and vertical 2
```

Comment:
@efischer19 expressed terror at the thought of trying to diagram it, but I could give it a shot.